$OpenBSD: patch-gio_tests_file_c,v 1.8 2018/01/23 10:43:42 mpi Exp $

The current kqueue backend does not monitor non-existing file correctly.

--- gio/tests/file.c.orig	Wed Aug 19 05:35:30 2015
+++ gio/tests/file.c	Wed Sep 23 09:25:35 2015
@@ -450,11 +450,12 @@ test_create_delete (gconstpointer d)
   data->monitor = g_file_monitor_file (data->file, 0, NULL, &error);
   g_assert_no_error (error);
 
-  /* This test doesn't work with GPollFileMonitor, because it assumes
-   * that the monitor will notice a create immediately followed by a
-   * delete, rather than coalescing them into nothing.
+  /* This test doesn't work with GPollFileMonitor nor GKqueueFileMonitor,
+   * because it assumes that the monitor will notice a create immediately
+   * followed by a delete, rather than coalescing them into nothing.
    */
-  if (!strcmp (G_OBJECT_TYPE_NAME (data->monitor), "GPollFileMonitor"))
+  if ((!strcmp (G_OBJECT_TYPE_NAME (data->monitor), "GPollFileMonitor")) ||
+      (!strcmp (G_OBJECT_TYPE_NAME (data->monitor), "GKqueueFileMonitor")))
     {
       g_test_skip ("skipping test for this GFileMonitor implementation");
       goto skip;
